src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ReferenceNumberGeneratorService.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.companieshouse.api.strikeoffobjections.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IReferenceNumberGeneratorService;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Service
+public class ReferenceNumberGeneratorService implements IReferenceNumberGeneratorService {
+
+    private static final String REFERENCE_NUMBER_PREFIX = "OBJ-";
+
+    private final Supplier<LocalDateTime> dateTimeSupplier;
+
+    @Autowired
+    public ReferenceNumberGeneratorService(Supplier<LocalDateTime> dateTimeSupplier) {
+        this.dateTimeSupplier = dateTimeSupplier;
+    }
+
+    /**
+     * Generates a 12 digit reference number based on the day and a random uuid, and inserts a hyphen every 4 characters
+     * e.g. OBJ-1F3C-A2E4-5D6B
+     * @return a 12 digit reference with hyphens every 4 digits, prefixed with REFERENCE_NUMBER_PREFIX
```

Comment:
If you replace REFERENCE_NUMBER_PREFIX with {@value #REFERENCE_NUMBER_PREFIX} the generated Javadoc will handle it nicely (display constant name but link to actual value).